torch/_tensor_docs.py

```diff
@@ -3,135 +3,159 @@
 import torch._C
 from torch._C import _add_docstr as add_docstr
 
-add_docstr(torch._C.FloatTensorBase.abs,
-           """
-abs() -> Tensor
+
+tensor_classes = [
+    'DoubleTensorBase',
+    'FloatTensorBase',
+    'LongTensorBase',
+    'IntTensorBase',
+    'ShortTensorBase',
+    'CharTensorBase',
+    'ByteTensorBase',
+]
+
+
+def add_docstr_all(method, docstr):
+    for cls_name in tensor_classes:
+        cls = getattr(torch._C, cls_name)
+        try:
+            add_docstr(getattr(cls, method),
+                       docstr.format(
+                           cls_name.replace('Base', ''),
+                           cls_name.replace('TensorBase', '')))
+        except AttributeError:
+            pass
+
+
+add_docstr_all('abs',
+               """
+abs() -> {0}
 
 See :func:`torch.abs`
 """)
 
-add_docstr(torch._C.FloatTensorBase.abs_,
-           """
-abs_() -> Tensor
+add_docstr_all('abs_',
+               """
+abs_() -> {0}
 
 In-place version of :meth:`~Tensor.abs`
 """)
 
-add_docstr(torch._C.FloatTensorBase.acos,
-           """
-acos() -> Tensor
+add_docstr_all('acos',
+               """
+acos() -> {0}
 
 See :func:`torch.acos`
 """)
 
-add_docstr(torch._C.FloatTensorBase.acos_,
-           """
-acos_() -> Tensor
+add_docstr_all('acos_',
+               """
+acos_() -> {0}
```

Comment:
So if you look at the [online docs](http://pytorch.org/docs/), you'll see that we have them for a single thing called `torch.Tensor`, so we don't repeat them for each type. This is why these return types were like that, and I think it's better to keep them that way